app/src/main/java/com/internship/move/data/dto/UserApi.kt

```diff
@@ -0,0 +1,19 @@
+package com.internship.move.data.dto
+
+import com.internship.move.data.model.User
+import retrofit2.http.Body
+import retrofit2.http.Field
+import retrofit2.http.GET
+import retrofit2.http.POST
+
+interface UserApi {
+
+    @GET("/user/users")
+    suspend fun getUsers(): List<User>
+
+    @POST("/user/register")
+    suspend fun register(@Body user: User): User
```

Comment:
Create another class, let's call it `UserResponse` with the sole purpose of getting the response back from the server.

Then map the response to the `User` object.
An object should have only 1 duty.